google-cloud-bigquery/acceptance/bigquery/advanced_test.rb

```diff
@@ -0,0 +1,167 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "bigquery_helper"
+
+describe Google::Cloud::Bigquery, :advanced, :bigquery do
+  let(:dataset_id) { "#{prefix}_dataset" }
+  let(:dataset) { bigquery.dataset(dataset_id) || bigquery.create_dataset(dataset_id) }
+  let(:table_id) { "examples_table" }
+  let(:table) { @table }
+
+  before do
+    @table = get_or_create_example_table dataset, table_id
+  end
+
+  it "loads and returns the properly formatted data" do
+    table.data.sort_by { |r| r["id"] }.zip(example_table_rows) do |returned_row, example_row|
```

Comment:
I see interesting. 

Is it correct for me to group this together in my head as such?

```rb
table.data.sort_by( { |r| r["id"]} ).zip(example_table_rows) ...